### ISSUE TITLE
Formula.Firewall: Query 'Events' (ste 'Renamed Columns9') 

### ISSUE THREAD

**Issue by Pieter-Veenstra** [SharePoint/sp-dev-gdpr-activity-hub#33]
Within PowerBI  I'm getting the following error:

~~~~
Formula.Firewall: Query 'Events' (step 'Renamed Columns9') references other queries or steps, so it may not directly access a data source. Please rebuild this data combination.
~~~~

**Comment by PaoloPia** [comment]
Which version of Power BI are you using?

**Comment by Pieter-Veenstra** [comment]
I am using all the latest as I had a clean environment last week

**Comment by tomkell29** [comment]
I have the same problem.
Using the latest version of Power BI Desktop (2.52.4921.581 64-bit November 2017)

I see a gold bar underneath the ribbon saying 'There are pending changes in your queries that haven't been applied'. As soon as I click on apply buttons, it tries to load the data, but then comes up with the following:

Events
Formula.Firewall: Query 'Events' (step 'Renamed Columns9') references other queries or steps, so it may not directly access a data source. Please rebuild this data combination.

The other items just say 'Load was cancelled by an error in loading a previous table'.

The same happens when clicking on the 'Refresh' button.

I've checked permissions on the site and I've ensured that I'm signed in to Power BI.

**Comment by westerdaled** [comment]
@tomkell29  I have opened the .pbit file multiple times incl viewing the issues in the advanced editor but I am still get passed the error as you have first described above.   I don't see how simply re-opening the .pbit solved the issue - tho I am glad it worked for you.

**Comment by tomkell29** [comment]
Hi @westerdaled,

I'm not sure how re-opening fixed it for me either, but somehow it did - perhaps it was just a permissions thing.

Have you entered some dummy data on your GDPR SharePoint site? That may also help and is mentioned in the instructions (although not very clear and I missed it at first) - [Setup Guide](https://github.com/SharePoint/sp-dev-gdpr-activity-hub/blob/master/Documentation/Setup-Guide.md)

**Comment by westerdaled** [comment]
@tomkell29   Yep adding test data resolved this issue then re-opening  This mandatory step needs  be made clearer I guess.

**Comment by tomkell29** [comment]
No problemo! Glad to hear you got it working, @westerdaled. 

@PaoloPia, perhaps you could amend the instructions to make this step clearer?

**Comment by PaoloPia** [comment]
Hi @westerdaled and @tomkell29,
Thanks for your feedback. Actually, we already add one sample item per list (Events and Request) during the provisioning phase. Did you add more than that? Thanks.

**Comment by westerdaled** [comment]
@PaoloPia  

I think @tomkell29  has a point as neither of us attached too much importance to this part of the instructions:

> "Notice that it is highly suggested to load some data in the SharePoint Online site (eventually fake data that you can delete at the end of this section) in order to allow the dashboard to find all the reference items in the data source site

When we both entered some data, the BI dashboards lit up and you can see how great your solution is!  Indeed I  may even reinstall for the client I am now working with.
